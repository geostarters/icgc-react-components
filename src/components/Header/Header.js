// @flow

import React from "react";
import PropTypes from "prop-types";
import { Image } from "semantic-ui-react";

import ResizeComponent from "../../hoc/ResizeComponent/ResizeComponent";

import IMAGES from "../../resources/images";
import styles from "./Header.module.css";

const Header = ({title = "Nou prototip", customStyleBar = {}, pathLogo = IMAGES.ICGC, width, showRightElements, children, logoSize = "small", customStyleLogo}) => {

	const isMobile = width <= 500;

	return (
		<div className={styles.containerHeader} style={customStyleBar}>

			{ !isMobile && 
				<div className={styles.containerlogo} style={customStyleLogo}>
					<Image src={pathLogo} size={logoSize}/>
				</div>
			}

			<div className={styles.containertitle}>
				<h2>{title}</h2>
			</div>

			{showRightElements &&
				<div className={styles.containeritemRight}>
					{children}
				</div>
			}

		</div>
	);

};

// eslint-disable-next-line new-cap
export default React.memo(ResizeComponent(Header));

Header.propTypes = {
	title: PropTypes.string,
	pathLogo: PropTypes.string,
	logoSize: PropTypes.string,
	showModalInfo: PropTypes.bool,
	width: PropTypes.number,
	customStyleBar: PropTypes.object,
	customStyleLogo: PropTypes.object,
	children: PropTypes.object,
	showRightElements: PropTypes.bool
};
