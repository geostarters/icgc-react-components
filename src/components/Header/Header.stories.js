//@flow

import React from "react";
import LogoICGC from "../LogoICGC/LogoICGC";

import Header from "./Header";

import IMAGES from "../../resources/images";

import "semantic-ui-css/semantic.min.css";

export default {
	title: "Header",
	component: Header
};

const Template = args => (<Header {...args}/>);

export const _Header = Template.bind({});
_Header.storyName = "Header";

export const HeaderIcgc = Template.bind({});
HeaderIcgc.storyName = "Header logo ICGC white";
HeaderIcgc.args = {
	logoSize: "medium",
	customStyleBar: {backgroundColor: "#0064af", color: "#fff"},
	pathLogo: IMAGES.ICGC_white
};

export const Header2Logos = args => (<Header {...args}>
	<div>
		<LogoICGC />
	</div>
</Header>);
Header2Logos.storyName = "Header 2 logos";
Header2Logos.args = {
	title:"VERSIÃ“ BETA",
	pathLogo: "https://eines.icgc.cat/contextmaps/static/media/CONTEXTMAPS_LOGO_2019.dc127f4a.svg",
	logoSize: "medium",
	customStyleBar: {backgroundColor: "#0064af", color: "#fff"},
	showRightElements: true,
	customStyleLogo: {maxWidth: "15em"}
};
