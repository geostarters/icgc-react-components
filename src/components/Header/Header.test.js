// @flow

import React from "react";
import Header from "./Header";
import  { mount } from "enzyme";
import toJSON from "enzyme-to-json";

describe("Header Component", () => {

	it("renders correct properties", () => {

		const mockProps = {
			title: "Title",
			width: 1000,
			showRightElements: true
		};

		const component = mount(<Header {...mockProps}></Header>);

		expect(toJSON(component)).toMatchSnapshot();

	});

	it("renders without image for mobile", () => {

		const mockProps = {
			title: "Title",
			width: 400,
			showRightElements: false

		};
		const component = mount(<Header {...mockProps}></Header>);

		expect((component.find("img"))).toHaveLength(0);

	});

});
