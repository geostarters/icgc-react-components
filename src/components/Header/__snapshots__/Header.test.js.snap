// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correct properties 1`] = `
<ResizeComponent
  showRightElements={true}
  title="Title"
  width={1000}
>
  <Header
    showRightElements={true}
    title="Title"
    width={1000}
  >
    <div
      className="containerHeader"
      style={Object {}}
    >
      <div
        className="containerlogo"
      >
        <Image
          as="img"
          size="small"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "ICGC.svg",
            }
          }
          ui={true}
        >
          <img
            className="ui small image"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "ICGC.svg",
              }
            }
          />
        </Image>
      </div>
      <div
        className="containertitle"
      >
        <h2>
          Title
        </h2>
      </div>
      <div
        className="containeritemRight"
      />
    </div>
  </Header>
</ResizeComponent>
`;
